### PR TITLE
Support file uploads in homepage markdown widget

### LINKED ISSUE #1374: Add the ability to upload a file in the homepage plugin
The folder markdown widget supports this, but provides that folder as a target for the upload.

An upload target would need to be specified or provided by the plugin (perhaps a new private collection named "homepage", a new folder under that user's space, etc).

follow up to #1370.

### PR DESCRIPTION
**Pull request by fogleman**
Fixes #1374

@brianhelba

### CODE REVIEW

File: plugins/homepage/server/__init__.py [hunk 1]
```diff
@@ -36,14 +38,27 @@ def __init__(self):
...
     if event.info['key'] == KEY:
         event.preventDefault().stopPropagation()
 
 
+def getOrCreateAssetsFolder():
+    collection = ModelImporter.model('collection').createCollection(
+        NAME, public=False, reuseExisting=True)
+    folder = ModelImporter.model('folder').createFolder(
+        collection, NAME, parentType='collection', public=False,
+        reuseExisting=True)
```
Comment (zachmullen) [inline plugins/homepage/server/__init__.py]: Shouldn't the folder at least be public so that everyone can see the images?

File: plugins/homepage/server/__init__.py [hunk 2]
```diff
@@ -21,8 +21,10 @@
 from girder.api import access
 from girder.api.describe import Description, describeRoute
 from girder.api.rest import Resource
+from girder.utility.model_importer import ModelImporter
 
 KEY = 'homepage.markdown'
+NAME = 'Homepage Assets'
```
Comment (brianhelba) [inline plugins/homepage/server/__init__.py]: These are typically in a `constants.py` file, as members of a class. It's probably not as big of a deal for a plugin this size.

File: plugins/homepage/server/__init__.py (hunk 1 above)
Comment (brianhelba) [inline plugins/homepage/server/__init__.py]: Indeed. Make the collection private and the folder public. Then everyone can download from it without seeing it in their listing.

File: girder/models/collection.py [hunk 3]
```diff
@@ -114,22 +115,33 @@ def createCollection(self, name, creator, description='', public=True):
...
+            if existing:
+                return existing
+
         now = datetime.datetime.utcnow()
 
         collection = {
             'name': name,
             'description': description,
-            'creatorId': creator['_id'],
+            'creatorId': creator['_id'] if creator else None,
```
Comment (zachmullen) [inline girder/models/collection.py]: This introduces the notion of collections with no creator, so we'll definitely want to add some testing since that's a new feature.

File: plugins/homepage/server/__init__.py (hunk 1 above)
Comment (fogleman) [inline plugins/homepage/server/__init__.py]: Oops, fixing.

**Comment by brianhelba** [comment]
Feature-wise, this LGTM. The collection having no creator needs to be resolved one way or another (as a supported feature with testing, or by making it owned by some arbitrary site admin).

**Comment by fogleman** [comment]
Doesn't look like collection creator is really used for much of anything. I can add a test making sure the creator can now be `None`, but not sure what else to test?

**Comment by brianhelba** [comment]
I share @zachmullen hesitation about the implications of allowing collections to have no creator, but I'll let him make the call on this.

**Comment by zachmullen** [comment]
The alternative is to have the asset folder be a configurable system setting and only allow uploading in the widget if the value is set

**Comment by fogleman** [comment]
Folders can also have no creator, that was already the case. I don't see any real implications.

**Comment by zachmullen** [comment]
I'm honestly not sure what assumptions this might break. It clearly doesn't break anything just by calling `createCollection`, if a problem exists it would come up somewhere else where the collection was being passed in and someone was assuming it has a creatorId. It's pretty hard to figure out how to test that, though, so I guess it's OK as is.

**Comment by fogleman** [comment]
LGTY?

**Comment by zachmullen** [comment]
LGTM
